File: Matlab/prune.m
function new_theta = prune(theta,theta_plus,theta_d)

% a = 0.1;
% b = 0.7;

a = [1,0.1];
b = [4,1];

new_theta =theta;
dist = inf;

            for ink = 1:length(new_theta(1,:))
                new_dist = sum(abs(theta_plus-new_theta(:,ink))./(b-a)');
                dist = min(dist,new_dist);
            end
            up = sign(min(theta_plus-a'));
            down  = sign(min(b'-theta_plus));
             if (dist>theta_d)&&(up>-1)&&(down>-1)
                new_theta = [new_theta,theta_plus];   
            end
           
%             if (dist>theta_d)&&(a(1)<=theta_plus(1))&&(theta_plus(1)<=b(1))&&(a(2)<=theta_plus(2))&&(theta_plus(2)<=b(2))
%                 new_theta = [new_theta,theta_plus];   
%             end         

File: R/prune.r
prune <- function(theta, theta_plus, theta_d){
    a = c(1,0.1)
    b = c(4, 1)

    #R parameters are passed by value, not need to copy

    dist = Inf

    for( col in 1:nrow(theta)){
        new_dist = sum(abs(theta_plus - theta[,col]/(b-a)))
        dist = min(dist, new_dist)
    }
    up = sign(pmin(theta_plus-a))
    down = sign(pmin(b-theta_plus))
    if(dist>theta_d && up > -1  && down > -1 ){
        theta = cbind(theta, theta_plus)
    }

    return(theta)

}